### ISSUE TITLE
[Ceph] Unspecified filestore osds are created on operator recreation

### ISSUE THREAD

**Issue by NickUfer** [rook/rook#2445]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

Not specified filestore osds are created

cephclusters.ceph.rook.io rook-ceph:
```
spec:
  cephVersion:
    image: ceph/ceph:v13.2.2-20181023
  dashboard:
    enabled: true
  dataDirHostPath: /var/lib/rook
  mon:
    allowMultiplePerNode: false
    count: 3
  network:
    hostNetwork: false
  rbdMirroring:
    workers: 0
  storage:
    config:
      databaseSizeMB: "1024"
      journalSizeMB: "1024"
      storeType: bluestore
    nodes:
    - config:
        storeType: bluestore
      devices:
      - FullPath: ""
        config: null
        name: nvme1n1
      name: server01
      resources: {}
    - config:
        storeType: bluestore
      devices:
      - FullPath: ""
        config: null
        name: nvme1n1
      name: server02
      resources: {}
    - config:
        storeType: bluestore
      devices:
      - FullPath: ""
        config: null
        name: sda
      - FullPath: ""
        config: null
        name: sdb
      name: server03
      resources: {}
    useAllDevices: false
```

Instead of 4 osds 6 are created:

```
ID CLASS WEIGHT  TYPE NAME         STATUS REWEIGHT PRI-AFF 
-1       8.76755 root default                              
-3       7.27399     host server03                      
 0   hdd 3.63699         osd.0         up  1.00000 1.00000 
 2   hdd 3.63699         osd.2         up  1.00000 1.00000 
-2       0.90659     host server02                         
 4       0.44260         osd.4         up  1.00000 1.00000 
 1  nvme 0.46399         osd.1         up  1.00000 1.00000 
-4       0.58698     host server01                             
 5       0.12299         osd.5         up  1.00000 1.00000 
 3  nvme 0.46399         osd.3         up  1.00000 1.00000
```
OSDs 4 and 5 are unexpected

**Expected behavior:**

Respect of the spec.storage.useAllDevices directive

**How to reproduce it (minimal and precise):**

Specify bluestore OSDs and set spec.storage.useAllDevices = false

**Environment**:
* OS (e.g. from /etc/os-release):
```
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.1 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
* Kernel (e.g. `uname -a`):
`Linux server01 4.15.0-42-generic #45-Ubuntu SMP Thu Nov 15 19:32:57 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
* Cloud provider or hardware configuration: 
Bare Metal
* Rook version (use `rook version` inside of a Rook Pod):
```rook: v0.9.1```
* Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T21:04:45Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T20:56:12Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
```

* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/Rook/master/toolbox.html)): 
HEALTH_OK

**Comment by clkao** [comment]
The problem is [Provision using deviceConfigured as deviceSpecified]( https://github.com/rook/rook/blob/a85aa847198663fba38b38953eb6c381a5b05cde/pkg/daemon/ceph/osd/daemon.go#L136) incorrectly. when a no-op orchestration is triggered where all devices are configured, the provisioner runs the node as if the user did not specify any device and wanted to use the default hostDir.

**Comment by travisn** [comment]
@NickUfer @jmaerki @clkao I tried to repro, but don't see another OSD created yet. What am I missing?

1. Create the cluster with these osd settings:
```
  storage: 
    useAllNodes: false
    useAllDevices: false
    nodes:
    - name: "minikube"
      devices:
      - name: "sdc"
```
2. Wait for the osd to start and the operator log shows the orchestration is completed
3. Restart the operator

**Comment by NickUfer** [comment]
I just updated the ceph version to `v13.2.4-20190109`. Now on server01 with my specified nvme1n1 as a bluestore OSD it created a new filestore OSD, which is not specified, and it didn't created the specified osd on nvme1n1.
I don't know what's happening here :D

**Comment by clkao** [comment]
@NickUfer can you paste the logs from the prepare-server* job pod?

**Comment by clkao** [comment]
@travisn can you try triggering an re-orchestration by setting osd resources?

**Comment by NickUfer** [comment]
Okay I think I now know why it created the filestore OSDs. `server01` is a special one...it is the only server where `nvme0n1` is the OSD instead of `nvme1n1` and I forgot that when I changed some things.
So the mistake was that I defined a wrong OSD.

But I think it is still a bug that when you specify a wrong OSD and so the device list with all found drives is empty it creates a filestore OSD (or even more on every drive one)
